### PR TITLE
Improve reduce function for Automotive Scanner Enumerators

### PR DESCRIPTION
**Pull request by polybassa**

### CODE REVIEW

File: scapy/contrib/automotive/scanner/enumerator.py [hunk 1]
```diff
@@ -173,12 +172,14 @@ def _get_initial_requests(self, **kwargs):
     def __reduce__(self):  # type: ignore
         f, t, d = super(ServiceEnumerator, self).__reduce__()  # type: ignore
         try:
-            del d["_request_iterators"]
+            for k, v in d["_request_iterators"].items():
```
Comment (p-l-) [inline scapy/contrib/automotive/scanner/enumerator.py:175]:
```suggestion
            for k, v in six.iteritems(d["_request_iterators"]):
```

File: scapy/contrib/automotive/scanner/enumerator.py [hunk 2]
```diff
@@ -173,12 +172,14 @@ def _get_initial_requests(self, **kwargs):
...
+                d["_request_iterators"][k] = list(v)
         except KeyError:
             pass
 
         try:
-            del d["_retry_pkt"]
+            for k in d["_retry_pkt"].keys():
```
Comment (p-l-) [inline scapy/contrib/automotive/scanner/enumerator.py:181]:
```suggestion
            for k in d["_retry_pkt"]:
```

File: test/contrib/automotive/scanner/uds_scanner.uts [hunk 3]
```diff
@@ -128,21 +128,21 @@ s = EcuState(session=1)
 
 assert False == e._evaluate_response(s, UDS(b"\x27\x01"), None, **config)
 config = {"exit_if_service_not_supported": True}
-assert e._retry_pkt[s] == None
+assert e._retry_pkt[s] == []
```
Comment (p-l-) [inline test/contrib/automotive/scanner/uds_scanner.uts:131]: For consistency
```suggestion
assert not e._retry_pkt[s]
```

File: test/contrib/automotive/scanner/uds_scanner.uts [hunk 4]
```diff
@@ -128,21 +128,21 @@ s = EcuState(session=1)
...
 assert True == e._evaluate_response(s, UDS(b"\x27\x01"), UDS(b"\x7f\x27\x11"), **config)
-assert e._retry_pkt[s] == None
+assert e._retry_pkt[s] == []
```
Comment (p-l-) [inline test/contrib/automotive/scanner/uds_scanner.uts:133]:
```suggestion
assert not e._retry_pkt[s]
```

File: test/contrib/automotive/scanner/uds_scanner.uts [hunk 5]
```diff
@@ -128,21 +128,21 @@ s = EcuState(session=1)
...
 assert True == e._evaluate_response(s, UDS(b"\x27\x01"), UDS(b"\x7f\x27\x24"), **config)
 assert e._retry_pkt[s] == UDS(b"\x27\x01")
 assert False == e._evaluate_response(s, UDS(b"\x27\x02"), UDS(b"\x7f\x27\x24"), **config)
-assert e._retry_pkt[s] is None
+assert e._retry_pkt[s] == []
```
Comment (p-l-) [inline test/contrib/automotive/scanner/uds_scanner.uts:137]:
```suggestion
assert not e._retry_pkt[s]
```

File: test/contrib/automotive/scanner/uds_scanner.uts [hunk 6]
```diff
@@ -128,21 +128,21 @@ s = EcuState(session=1)
...
 assert True == e._evaluate_response(s, UDS(b"\x27\x01"), UDS(b"\x7f\x27\x37"), **config)
 assert e._retry_pkt[s] == UDS(b"\x27\x01")
 assert False == e._evaluate_response(s, UDS(b"\x27\x01"), UDS(b"\x7f\x27\x37"), **config)
-assert e._retry_pkt[s] == None
+assert e._retry_pkt[s] == []
```
Comment (p-l-) [inline test/contrib/automotive/scanner/uds_scanner.uts:141]:
```suggestion
assert not e._retry_pkt[s]
```

File: test/contrib/automotive/scanner/uds_scanner.uts [hunk 7]
```diff
@@ -128,21 +128,21 @@ s = EcuState(session=1)
...
 assert True == e._evaluate_response(s, UDS(b"\x27\x01"), UDS(b"\x67\x01ab"), **config)
-assert e._retry_pkt[s] == None
+assert e._retry_pkt[s] == []
```
Comment (p-l-) [inline test/contrib/automotive/scanner/uds_scanner.uts:143]:
```suggestion
assert not e._retry_pkt[s]
```

File: test/contrib/automotive/scanner/uds_scanner.uts [hunk 8]
```diff
@@ -128,21 +128,21 @@ s = EcuState(session=1)
...
 assert False == e._evaluate_response(s, UDS(b"\x27\x01"), UDS(b"\x67\x02ab"), **config)
-assert e._retry_pkt[s] == None
+assert e._retry_pkt[s] == []
```
Comment (p-l-) [inline test/contrib/automotive/scanner/uds_scanner.uts:145]:
```suggestion
assert not e._retry_pkt[s]
```

**Review by p-l- — CHANGES_REQUESTED** [review]
Apart for those tiny changes, LGTM

**Comment by polybassa** [comment]
Thanks a lot for your review!

**Comment by p-l-** [comment]
Tests failures seem unrelated, re-running the failed tests

**Review by p-l- — APPROVED** [review]

**Pull request merged**